reference/origin_direct.go

```diff
@@ -0,0 +1,35 @@
+package reference
+
+import (
+	"github.com/hashicorp/hcl-lang/lang"
+	"github.com/hashicorp/hcl/v2"
+)
+
+// DirectOrigin represents an origin which directly targets a file
+// and doesn't need a matching target
+type DirectOrigin struct {
+	// Range represents a range of a local traversal or an attribute
+	Range hcl.Range
+
+	// TargetPath represents what Path does the origin targets
```

Comment:
```suggestion
	// TargetPath represents what (directory) Path does the origin targets
```
Nitpick - just to clarify this is not a file path